src/core/strategies/token.js

```diff
@@ -0,0 +1,285 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018-2023 Open Assessment Technologies SA ;
+ */
+
+import { terms } from '../terms.js';
+import tokensHelper from '../tokens.js';
+
+/**
+ * List of tokens that refuse explicit positive sign
+ * @type {string[]}
+ */
+const refuseExplicitPositive = ['LPAR', 'SUB', 'ADD', 'MUL', 'DIV', 'MOD', 'POW', 'ASSIGN'];
+
+/**
+ * Checks if a token accept an explicit positive sign on the right
+ * @param {token} token
+ * @returns {boolean}
+ */
+const acceptExplicitPositive = token =>
+    !token || (!tokensHelper.isFunction(token) && refuseExplicitPositive.indexOf(token.type) === -1);
+
+/**
+ * Produces a descriptor to insert a negative sign
+ * @param {token} token
+ * @returns {tokenChange}
+ */
+function insertNegativeSign(token) {
+    const { offset } = token;
+    const { value } = terms.SUB;
+    return {
+        offset,
+        length: 0,
+        value,
+        move: value.length
+    };
+}
+
+/**
+ * Produces a descriptor to replace a sign by a negative sign
+ * @param {token} token
+ * @returns {tokenChange}
+ */
+function replaceByNegativeSign(token) {
+    const { offset } = token;
+    const { value } = terms.SUB;
+    return {
+        offset,
+        length: token.value.length,
+        value,
+        move: value.length - token.value.length
```

Comment:
i can see on this expression you use value, which is defined here, but also token. could it be better to use offset.value.length? (im not sure of the structure of offset but looks like a copy of token)

Reply:
`offset` is a member of `token`, this instruction is ES6...

`const { offset } = token` is equivalent of `const offset = token.offset`